pkg/cli/interactive_tests/common.tcl

```diff
@@ -88,6 +88,44 @@ proc interrupt {} {
     sleep 0.4
 }
 
+# Convenience function that sends Ctrl+C and waits more reliably for process to exit
+# This uses a retry approach similar to stop_server but adapted for expect environment
+proc interrupt_and_wait {} {
+    report "INTERRUPT AND WAIT FOR FOREGROUND PROCESS"
+    
+    # Send the interrupt signal
+    send "\003"
+    
+    # Use a retry loop similar to stop_server's wait mechanism
+    # Try up to 30 times with 1 second intervals, like stop_server does
+    for {set i 1} {$i <= 30} {incr i} {
+        # Try to expect the shell prompt with a short timeout (1 second per attempt)
+        set timeout 1
+        expect {
+            ":/# " {
+                report "FOREGROUND PROCESS EXITED SUCCESSFULLY"
+                # Restore original timeout
+                set timeout 45
```

Comment:
Nit: you could wrap the loop body into try/finally and restore the timeout in one place, i.e., in finally.